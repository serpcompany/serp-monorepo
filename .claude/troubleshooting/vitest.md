https://vitest.dev/guide/common-errors.html