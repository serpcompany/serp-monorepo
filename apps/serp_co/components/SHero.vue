<template>
  <div>
    <div class="py-24">
      <div>
        <div class="items-center">
          <div
            class="flex flex-col items-center text-center lg:items-start lg:text-left"
          >
            <h1 class="my-4 text-2xl font-bold text-pretty lg:text-5xl">
              {{ headline }}
            </h1>
            <p
              v-if="subheadline"
              class="text-muted-foreground mb-8 max-w-2xl lg:text-xl"
            >
              {{ subheadline }}
            </p>
            <div
              class="flex w-full flex-col justify-center gap-2 sm:flex-row lg:justify-start"
            >
              <div v-if="showSearchBar">
                <!-- <SearchBar /> -->
              </div>
              <template v-if="showButtons">
                <u-button
                  label="Primary Button"
                  class="w-full cursor-pointer rounded-none text-lg sm:w-auto"
                  >Primary Button</u-button
                >
                <u-button
                  variant="outline"
                  class="w-full cursor-pointer rounded-none text-lg sm:w-auto"
                >
                  Buying Advice
                </u-button>
              </template>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
interface Props {
  headline: string;
  subheadline?: string;
  showSearchBar?: boolean;
  showButtons?: boolean;
}

withDefaults(defineProps<Props>(), {
  headline: '',
  subheadline: undefined,
  showSearchBar: true,
  showButtons: true
});
</script>
