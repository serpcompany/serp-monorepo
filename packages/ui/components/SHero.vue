<script setup lang="ts">
  interface Props {
    headline: string;
    subheadline?: string;
    showSearchBar?: boolean;
    showButtons?: boolean;
  }

  withDefaults(defineProps<Props>(), {
    headline: '',
    subheadline: undefined,
    showSearchBar: true,
    showButtons: true,
  });
</script>

<template>
  <div class="py-20">
    <div class="mx-auto max-w-5xl px-4 sm:px-6 lg:px-8">
      <div
        class="flex flex-col items-center gap-4 text-center lg:items-start lg:text-left"
      >
        <h1 class="text-3xl font-bold text-pretty lg:text-5xl">
          {{ headline }}
        </h1>

        <template v-if="subheadline">
          <p class="text-muted max-w-2xl lg:text-xl">
            {{ subheadline }}
          </p>
        </template>

        <template v-if="showSearchBar && showButtons">
          <div
            class="mt-4 flex w-full flex-col justify-center gap-2 sm:flex-row lg:justify-start"
          >
            <div v-if="showSearchBar">
              <!-- <SearchBar /> -->
            </div>
            <template v-if="showButtons">
              <UButton
                label="Primary Button"
                class="w-full cursor-pointer rounded-none text-lg sm:w-auto"
              >
                Primary Button
              </UButton>
              <UButton
                variant="outline"
                class="w-full cursor-pointer rounded-none text-lg sm:w-auto"
              >
                Buying Advice
              </UButton>
            </template>
          </div>
        </template>
      </div>
    </div>
  </div>
</template>
