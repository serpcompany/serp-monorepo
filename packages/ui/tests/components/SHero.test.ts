import { describe, expect, it } from 'vitest';
import SHero from '../../components/SHero.vue';
import ComponentRender from '../componentRender';

describe('sHero Snapshot Tests', () => {
  it.each([
    // Default props (using component defaults)
    ['with default props', { props: {} }],
    // Only headline provided
    ['with headline only', { props: { headline: 'Welcome to Our Site' } }],
    // Headline with a subheadline
    [
      'with headline and subheadline',
      {
        props: {
          headline: 'Hello World',
          subheadline: 'This is a subheadline.',
        },
      },
    ],
    // Without search bar
    [
      'without search bar',
      { props: { headline: 'No Search', showSearchBar: false } },
    ],
    // Without buttons
    [
      'without buttons',
      { props: { headline: 'No Buttons', showButtons: false } },
    ],
    // All props customized
    [
      'with custom all props',
      {
        props: {
          headline: 'Custom Hero',
          subheadline: 'Amazing subheadline',
          showSearchBar: false,
          showButtons: false,
        },
      },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(`SHero ${desc}`, options, SHero);
      expect(html).toMatchSnapshot();
    },
  );
});
