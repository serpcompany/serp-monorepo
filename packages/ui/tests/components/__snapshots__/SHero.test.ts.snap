// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sHero Snapshot Tests > renders with custom all props correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      ">
        Custom Hero
      </h1>
      <p class="
        lg:text-xl
        max-w-2xl
        text-muted
      ">
        Amazing subheadline
      </p>
      <!-- v-if -->
    </div>
  </div>
</div>
`;

exports[`sHero Snapshot Tests > renders with default props correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      "></h1>
      <!-- v-if -->
      <div class="
        flex
        flex-col
        gap-2
        justify-center
        lg:justify-start
        mt-4
        sm:flex-row
        w-full
      ">
        <div>
          <!-- <SearchBar /> -->
        </div>
        <button
          class="
            aria-disabled:bg-primary
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            bg-primary
            cursor-pointer
            disabled:bg-primary
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:outline-2
            focus-visible:outline-offset-2
            focus-visible:outline-primary
            font-medium
            gap-1.5
            hover:bg-primary/75
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            sm:w-auto
            text-inverted
            text-lg
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Primary Button
          <!-- v-if -->
        </button>
        <button
          class="
            aria-disabled:bg-transparent
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            cursor-pointer
            dark:aria-disabled:bg-transparent
            dark:disabled:bg-transparent
            disabled:bg-transparent
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:ring-2
            focus-visible:ring-primary
            focus:outline-none
            font-medium
            gap-1.5
            hover:bg-primary/10
            inline-flex
            items-center
            px-2.5
            py-1.5
            ring
            ring-inset
            ring-primary/50
            rounded-none
            sm:w-auto
            text-lg
            text-primary
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Buying Advice
          <!-- v-if -->
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`sHero Snapshot Tests > renders with headline and subheadline correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      ">
        Hello World
      </h1>
      <p class="
        lg:text-xl
        max-w-2xl
        text-muted
      ">
        This is a subheadline.
      </p>
      <div class="
        flex
        flex-col
        gap-2
        justify-center
        lg:justify-start
        mt-4
        sm:flex-row
        w-full
      ">
        <div>
          <!-- <SearchBar /> -->
        </div>
        <button
          class="
            aria-disabled:bg-primary
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            bg-primary
            cursor-pointer
            disabled:bg-primary
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:outline-2
            focus-visible:outline-offset-2
            focus-visible:outline-primary
            font-medium
            gap-1.5
            hover:bg-primary/75
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            sm:w-auto
            text-inverted
            text-lg
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Primary Button
          <!-- v-if -->
        </button>
        <button
          class="
            aria-disabled:bg-transparent
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            cursor-pointer
            dark:aria-disabled:bg-transparent
            dark:disabled:bg-transparent
            disabled:bg-transparent
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:ring-2
            focus-visible:ring-primary
            focus:outline-none
            font-medium
            gap-1.5
            hover:bg-primary/10
            inline-flex
            items-center
            px-2.5
            py-1.5
            ring
            ring-inset
            ring-primary/50
            rounded-none
            sm:w-auto
            text-lg
            text-primary
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Buying Advice
          <!-- v-if -->
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`sHero Snapshot Tests > renders with headline only correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      ">
        Welcome to Our Site
      </h1>
      <!-- v-if -->
      <div class="
        flex
        flex-col
        gap-2
        justify-center
        lg:justify-start
        mt-4
        sm:flex-row
        w-full
      ">
        <div>
          <!-- <SearchBar /> -->
        </div>
        <button
          class="
            aria-disabled:bg-primary
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            bg-primary
            cursor-pointer
            disabled:bg-primary
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:outline-2
            focus-visible:outline-offset-2
            focus-visible:outline-primary
            font-medium
            gap-1.5
            hover:bg-primary/75
            inline-flex
            items-center
            px-2.5
            py-1.5
            rounded-none
            sm:w-auto
            text-inverted
            text-lg
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Primary Button
          <!-- v-if -->
        </button>
        <button
          class="
            aria-disabled:bg-transparent
            aria-disabled:cursor-not-allowed
            aria-disabled:opacity-75
            cursor-pointer
            dark:aria-disabled:bg-transparent
            dark:disabled:bg-transparent
            disabled:bg-transparent
            disabled:cursor-not-allowed
            disabled:opacity-75
            focus-visible:ring-2
            focus-visible:ring-primary
            focus:outline-none
            font-medium
            gap-1.5
            hover:bg-primary/10
            inline-flex
            items-center
            px-2.5
            py-1.5
            ring
            ring-inset
            ring-primary/50
            rounded-none
            sm:w-auto
            text-lg
            text-primary
            transition-colors
            w-full
          "
          type="button"
        >
          <!-- v-if -->
          Buying Advice
          <!-- v-if -->
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`sHero Snapshot Tests > renders without buttons correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      ">
        No Buttons
      </h1>
      <!-- v-if -->
      <!-- v-if -->
    </div>
  </div>
</div>
`;

exports[`sHero Snapshot Tests > renders without search bar correctly 1`] = `
<div class="py-20">
  <div class="
    lg:px-8
    max-w-5xl
    mx-auto
    px-4
    sm:px-6
  ">
    <div class="
      flex
      flex-col
      gap-4
      items-center
      lg:items-start
      lg:text-left
      text-center
    ">
      <h1 class="
        font-bold
        lg:text-5xl
        text-3xl
        text-pretty
      ">
        No Search
      </h1>
      <!-- v-if -->
      <!-- v-if -->
    </div>
  </div>
</div>
`;
